plugins/source/aws/README.md

```diff
@@ -0,0 +1,329 @@
+so, following the principle of least privilege, it's recommended to grant it read-only permissions.
+
+There are multiple ways to authenticate with AWS, and CloudQuery respects the AWS credential provider chain. This means that CloudQuery will follow the following priorities when attempting to authenticate:
+
+- The `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` environment variables.
+- The `credentials` and `config` files in `~/.aws` (the `credentials` file takes priority).
+  - You can also use `aws sso` to authenticate cloudquery - [you can read more about it here](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html).
+- IAM roles for AWS compute resources (including EC2 instances, Fargate and ECS containers).
+
+You can read more about AWS authentication [here](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials) and [here](https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html).
+
+### Environment Variables
+
+CloudQuery can use the credentials from the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and
+`AWS_SESSION_TOKEN` environment variables (`AWS_SESSION_TOKEN` can be optional for some accounts). For information on obtaining credentials, see the
+[AWS guide](https://aws.github.io/aws-sdk-go-v2/docs/getting-started/#get-your-aws-access-keys).
+
+To export the environment variables (On Linux/Mac - similar for Windows):
+
+```bash
+export AWS_ACCESS_KEY_ID={Your AWS Access Key ID}
+export AWS_SECRET_ACCESS_KEY={Your AWS secret access key}
+export AWS_SESSION_TOKEN={Your AWS session token}
+```
+
+### Shared Configuration files
+
+CloudQuery can use credentials from your `credentials` and `config` files in the `.aws` directory in your home folder.
+The contents of these files are practically interchangeable, but CloudQuery will prioritize credentials in the `credentials` file.
+
+For information about obtaining credentials, see the
+[AWS guide](https://aws.github.io/aws-sdk-go-v2/docs/getting-started/#get-your-aws-access-keys).
+
+Here are example contents for a `credentials` file:
+
+```toml title="~/.aws/credentials"
+[default]
+aws_access_key_id = <YOUR_ACCESS_KEY_ID>
+aws_secret_access_key = <YOUR_SECRET_ACCESS_KEY>
+```
+
+You can also specify credentials for a different profile, and instruct CloudQuery to use the credentials from this profile instead of the default one.
+
+For example:
+
+```toml title="~/.aws/credentials"
+[myprofile]
+aws_access_key_id = <YOUR_ACCESS_KEY_ID>
+aws_secret_access_key = <YOUR_SECRET_ACCESS_KEY>
+```
+
+Then, you can either export the `AWS_PROFILE` environment variable (On Linux/Mac, similar for Windows):
+
+```bash
+export AWS_PROFILE=myprofile
+```
+
+or, configure your desired profile in the `local_profile` field of your CloudQuery `aws.yml`:
+
+```yaml title="aws.yml"
+kind: source
+spec:
+  name: aws
+  spec:
+    accounts:
+      id: <account_alias>
+      local_profile: myprofile
```

Comment:
```suggestion
kind: "source"
spec:
  name: "aws"
  spec:
    accounts:
      id: "<account_alias>"
      local_profile: "<my_profile>"
```

When we generate the config we use quoted strings, so I'd keep the same format here. Also use the same convention for `spec.accounts`.
Also should this be:
```
      id: "<id>"
      local_profile: "<local_profile>"
```

Or

```
      account_alias: "<account_alias>"
      local_profile: "<local_profile>"
```